pkg/dogstatsd/replay/writer.go

```diff
@@ -333,8 +338,10 @@ func (tc *TrafficCaptureWriter) StopCapture() {
 // Enqueue enqueues a capture buffer so it's written to file.
 func (tc *TrafficCaptureWriter) Enqueue(msg *CaptureBuffer) bool {
 	qd := false
+	tc.RLock()
+	defer tc.RUnlock()
 
-	if tc.IsOngoing() {
+	if tc.accepting {
 		tc.Traffic <- msg
```

Comment:
There is still a deadlock in this code:
- A goroutine (1) executes `log.Debug("Capture shutting down")` and sleeps for a few CPU cycles.
- Another goroutine (2) executes line 341-345
-  goroutine (2) blocks on line 345 (`tc.Traffic <- msg`) if `tc.Traffic` is full. Goroutine (2) holds the mutex 
- goroutine (1) blocks on line 266 (`tc.Lock()`).

Note: It is unlikely to happen

Reply:
Yeah, turns out this happens pretty often when running our DogStatsD benchmark. Have fixed it.